fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -38,12 +39,131 @@
 UNEXPECTED_DATA_TYPE_ERROR_MESSAGE = "Unexpected data type {} encountered."
 ESTIMATOR_ERROR_MESSAGE = "'estimator' cannot be `None`."
 
-_SUPPORTED_CONSTRAINTS = [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS]
-
 logger = logging.getLogger(__name__)
 
+# All allowed utilities must have values between 0 and 1 and attain both extremes as
+# the threshold goes from -Inf to Inf.
+SUPPORTED_UTILITIES = {
+    'selection_rate',
+    'demographic_parity',
+    'false_positive_rate',
+    'false_negative_rate',
+    'true_positive_rate',
+    'true_negative_rate',
+}
+
+# For equalized odds, we only allow objectives that are non-decreasing in true_positives,
+# holding n, positives, negatives, true_negatives, and false_positives fixed.
+SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS = {
+    'accuracy_score',
+}
+
+SUPPORTED_CONSTRAINTS = [*SUPPORTED_UTILITIES, EQUALIZED_ODDS]
+NOT_SUPPORTED_CONSTRAINTS_ERROR_MESSAGE = (
+    "Currently only the following constraints are supported: {}.".format(
+        ", ".join(SUPPORTED_CONSTRAINTS)))
+NOT_SUPPORTED_OBJECTIVES_ERROR_MESSAGE = (
+    "Currently only the following objectives are supported: {}.".format(
+    ", ".join(METRIC_DICT.keys())))
+NOT_SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS_ERROR_MESSAGE = (
+    "Currently only the following objectives are supported: {}.".format(
+    ", ".join(SUPPORTED_OBJECTIVES_FOR_EQUALIZED_ODDS)))
+
+class InterpolatedThresholder(BaseEstimator, MetaEstimatorMixin):
+    """Binary predictor that thresholds continuous predictions of a base estimator.
+
+    At prediction time, the predictor takes as input both standard and sensitive features.
+    Based on the values of sensitive features, it then applies a randomized thresholding
+    transformation according to the provided `interpolation_dict`.
+    
+    :param estimator: base estimator
+    :param dict interpolation_dict: maps sensitive feature values to `Bunch` that describes the
```

Comment:
Is this a new way of writing the doc, or did you mean
```
:param interpolation_dict: …
:type interpolation_dict: dict
```
?

Reply:
this is supported by sphinx and looks more natural to me... i'm even thinking we should consider going numpy style like sklear, because it reads nicer when you call `help` in interactive python.

Reply:
@adrinjalali recently pointed that out in an issue, I think #314